src/lib/events/server/actions.ts

```diff
@@ -185,26 +198,49 @@ export const updateEvent: Action<{ slug: string }> = async (event) => {
     where: {
       slug: slug,
     },
-    select: { id: true },
+    select: { id: true, recurringParentId: true, startDatetime: true },
   });
   if (!existingEvent) {
     throw error(404, m.events_errors_eventNotFound());
   }
 
   if (image) eventData.imageUrl = await uploadImage(user, image, slug);
 
-  await prisma.event.update({
-    where: {
-      id: existingEvent.id,
-    },
-    data: {
-      ...eventData,
-      author: undefined,
-      tags: {
-        set: tags.map(({ id }) => ({ id })),
+  if (!isRecurring || editType === "THIS") {
+    await prisma.event.update({
+      where: {
+        id: existingEvent.id,
       },
-    },
-  });
+      data: {
+        ...eventData,
+        author: undefined,
+        tags: {
+          set: tags.map(({ id }) => ({ id })),
+        },
+      },
+    });
+  } else if (editType === "FUTURE") {
+    await prisma.event.updateMany({
+      where: {
+        recurringParentId: existingEvent.recurringParentId,
+        startDatetime: { gte: existingEvent.startDatetime },
+      },
+      data: {
+        ...eventData,
+        authorId: undefined,
+      },
```

Comment:
It doesn't seem possible to update relations with updatemany (https://github.com/prisma/prisma/issues/3143). We could update the tags but that would be a lot of db calls. 
Suggestions welcome!